packages/block-library/src/social-links/edit.js

```diff
@@ -58,23 +59,34 @@ export function SocialLinksEdit( props ) {
 
 	const {
 		iconBackgroundColorValue,
+		customIconBackgroundColor,
 		iconColorValue,
 		openInNewTab,
 		size,
 		layout,
 	} = attributes;
+
 	const usedLayout = layout || getDefaultBlockLayout( name );
 
-	// Remove icon background color if logos only style selected.
-	const logosOnly =
-		attributes.className?.indexOf( 'is-style-logos-only' ) >= 0;
+	// Remove icon background color when logos only style is selected or
+	// restore it when any other style is selected.
+	const logosOnly = attributes.className
+		? attributes.className.indexOf( 'is-style-logos-only' ) >= 0
+		: false;
 	useEffect( () => {
 		if ( logosOnly ) {
+			backgroundBackup.current = {
+				iconBackgroundColor,
+				iconBackgroundColorValue,
+				customIconBackgroundColor,
+			};
 			setAttributes( {
 				iconBackgroundColor: undefined,
 				customIconBackgroundColor: undefined,
 				iconBackgroundColorValue: undefined,
 			} );
+		} else {
+			setAttributes( { ...backgroundBackup.current } );
 		}
 	}, [ logosOnly, setAttributes ] );
```

Comment:
So this doesn't seem necessary (as mentioned before: https://github.com/WordPress/gutenberg/pull/34834#discussion_r715202668) but it may be better to have all the consumed variables in the hook dependencies and it doesn't end up adding much complexity.
```suggestion
			setAttributes( {
				iconBackgroundColor: undefined,
				customIconBackgroundColor: undefined,
				iconBackgroundColorValue: undefined,
			} );
			// Restore the attributes from backup on cleanup.
			return () => setAttributes( { ...backgroundBackup.current } );
		}
		// Backup icon background attributes.
		backgroundBackup.current = {
			iconBackgroundColor,
			iconBackgroundColorValue,
			customIconBackgroundColor,
		};
	}, [
		logosOnly,
		setAttributes,
		iconBackgroundColor,
		iconBackgroundColorValue,
		customIconBackgroundColor,
	] );
```

Reply:
Agreed.

Reply:
@stokesman I am happy that I get chance for contributing.